src/validation.cpp

```diff
@@ -3036,6 +3039,8 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
     if (!CheckBlockHeader(block, state, consensusParams, fCheckPOW))
         return false;
 
+    if (g_signet_blocks && fCheckPOW && block.GetHash() != consensusParams.hashGenesisBlock && !CheckBlockSolution(block, consensusParams)) return false;
+
```

Comment:
Perhaps do this in CheckBlockHeader?
That way unsigned blocks would be detected faster.
Also, I think the return false in a new line would be more readable.

Reply:
`CheckBlockSolution` requires the block, not just the header.